### PR TITLE
Add check for docs

### PR DESCRIPTION
**Pull request by domgreen**
This commit adds a check for any change to the documentation that has not been correctly committed. It does this by regenerating the docs in Travis and then seeing if there are any changes due for staging in git.

Have also added a CONTRIBUTING.md file so that it is easier for new contributors to understand the flow.

### CODE REVIEW

File: scripts/docs.sh [hunk 1]
```diff
@@ -22,6 +22,21 @@ function generate_markdown {
...
+}
+
+function check {
+    generate
+    count=$(git diff --numstat | wc -l | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')
+    echo $count
+    if [ "$count" = "0" ]; then
+        return 0
+    else
+        return 1
```
Comment (mwitkow) [inline scripts/docs.sh]: Can you make this `echo` useful information such as:
"Your markdown docs seem to be out of sync with the package docs. Please run `make` and consult `CONTRIBUTING.MD`"

File: .travis.yml [hunk 2]
```diff
@@ -15,7 +15,8 @@ install:
   - dep ensure
 
 script:
- - make test 
+ - make test
+ - make checkdocs
```
Comment (mwitkow) [inline .travis.yml]: Make sure the checkdocs are first.

**Comment by domgreen** [comment]
Also need to add `realpath`

**Comment by domgreen** [comment]
@mwitkow have made changes also note the annoying realpath change that I have added ... this is due to Travis not having realpath installed ... by the looks of things realpath is a can of worms.

**Comment by mwitkow** [comment]
Legit. You don't need real path though, there's a bash one liner that does that.

**Comment by domgreen** [comment]
Updated as per discussion to change to bash one liner 👍 merging

**Pull request merged**